src/details/ArborX_Ray.hpp

```diff
@@ -273,6 +274,58 @@ KOKKOS_INLINE_FUNCTION float overlapDistance(Ray const &ray,
   return (tmax - tmin);
 }
 
+// Möller–Trumbore intersection algorithm
+KOKKOS_INLINE_FUNCTION bool rayTriangleIntersect(Ray const &ray,
+                                                 Triangle const &triangle,
+                                                 float &t, float &u, float &v)
+{
+  auto const ab = makeVector(triangle.a, triangle.b);
+  auto const ac = makeVector(triangle.a, triangle.c);
+
+  auto const p = crossProduct(ray.direction(), ac);
+  auto const det = dotProduct(ab, p);
+
+  auto const epsilon = 0.0000001f;
+  // If the determinant is negative the triangle is back-facing.
+  // If the determinant is close to 0, the ray misses the triangle because they
+  // are in the same plane.
+  if (det > -epsilon && det < epsilon)
```

Comment:
Can we add more tests for this? I am thinking about have a series of similar triangles but in different scales.

Reply:
There is a problem with large and small triangles but the problem is not due to the `epsilon`. The problem is that we are using `float` and not `double` and so we accumulate round-off errors. The only solution is to use `double` everywhere. If we mix `float` and `double` some of the errors does not cancel each other and the result is worse. This means that `Vector` needs to use `double` instead of `float`. I don't know if it's something we want to do though.

Reply:
Can we print some warnings under certain conditions when the algorithm with float tends to have more misjudgment?

Reply:
It seems that the error in the dot product is a good indicator that there is a potential problem so we could do it. However, that means that we need to compare the dot product using `float` with the one using `double` so it's more costly. Also, there are cases where there is an error on the dot product but it doesn't change the final result.